eng/tools/typespec-validation/src/utils.ts

```diff
@@ -33,3 +34,25 @@ export function normalizePath(folder: string, path: PlatformPath = defaultPath)
     .join("/")
     .replace(/^([a-z]):/, (_match, driveLetter) => driveLetter.toUpperCase() + ":");
 }
+
+export async function gitDiffTopSpecFolder(host: TsvHost, folder: string) {
+  const git = host.gitOperation(folder);
+  let topSpecFolder = folder.replace(/(^.*specification\/[^\/]*)(.*)/, "$1");
+  let stdOutput = `Running git diff on folder ${topSpecFolder}`;
+  let gitStatus = await git.status(["--porcelain", topSpecFolder]);
+
+  let success = true;
+  let errorOutput: string | undefined;
+
+  if (!gitStatus.isClean()) {
+    success = false;
+    errorOutput = `Files generated: ${gitStatus.not_added}\n`;
+    errorOutput += `Files modified: ${gitStatus.modified}`;
+  }
+
+  return {
```

Comment:
Do we also need to undo the changes somewhere?

Reply:
the undo script is here: https://github.com/Azure/azure-rest-api-specs/blob/main/eng/scripts/Validate-TypeSpec.ps1#L27-L28
we're refraining from undoing it because we don't want devs to accidentally delete their local dev work every time they run tsv

Reply:
If we are splitting these checks into individual git diffs it would seem that one failing check will cause others to also fail if we don't undo between the steps. Perhaps we need an option for tsv to undo as it goes or if it should leave modified files.

Reply:
We are stopping on first occurrence of error. We think that additional rule execution will make the output more confusing. So I don't think we need to undo between steps